themes/default/content/docs/concepts/inputs-outputs/_index.md

```diff
@@ -13,16 +13,272 @@ aliases:
 
 ---
 
-Resource properties are treated specially in Pulumi, both for purposes of input and output.
+## Inputs
 
-All resource arguments accept *inputs*. Inputs are values of type {{< pulumi-input >}}, a type that permits either a raw value of a given type (such as string, integer, boolean, list, map, and so on), an asynchronously computed value (i.e., a `Promise` or `Task`), or an output read from another resource’s properties.
+All resources in Pulumi accept values that describe the way the resource behaves. We call these values *inputs*.
 
-All resource properties on the instance object itself are *outputs*. Outputs are values of type {{< pulumi-output >}}, which behave very much like [promises](https://en.wikipedia.org/wiki/Futures_and_promises). This is necessary because outputs are not fully known until the infrastructure resource has actually completed provisioning, which happens asynchronously. Outputs are also how Pulumi tracks dependencies between resources.
+{{< chooser language "javascript,typescript,python,go,csharp,java" >}}
+
+{{% choosable language javascript %}}
+
+```javascript
+// TODO
+```
+
+{{% /choosable %}}
+{{% choosable language typescript %}}
+
+```typescript
+const myId = new random.RandomId("mine", {
+    byteLength: 8, // byte_length is an input
+});
+```
+
+{{% /choosable %}}
+{{% choosable language python %}}
+
+```python
+my_id = random.RandomId("mine",
+    byte_length=8 # byte_length is an input
+)
+```
+
+{{% /choosable %}}
+{{% choosable language go %}}
+
+```go
+
+myId, err := random.NewRandomId(ctx, "mine", &random.RandomIdArgs{
+	ByteLength: pulumi.Int(8), // ByteLength is an input
+})
+if err != nil {
+	return err
+}
+
+```
+
+{{% /choosable %}}
+{{% choosable language csharp %}}
+
+```csharp
+var myId = new Random.RandomId("mine", new()
+{
+    ByteLength = 8, // ByteLength is an input
+});
+
+```
+
+{{% /choosable %}}
+{{% choosable language java %}}
+
+```java
+var myId = new RandomId("mine", RandomIdArgs.builder()
+    .byteLength(8) // byteLength is an input
+    .build());
+
+```
+
+{{% /choosable %}}
+{{< /chooser >}}
+
+_Inputs_ are generally representations of the parameters to the underlying API call of any resource that Pulumi is managing.
```

Comment:
```suggestion
_Inputs_ are the parameters to the resource that Pulumi is managing.
```

Simplify language here, we have providers like Command, Docker, etc., that make this unclear, and component providers muddy the water.

Reply:
This was the main reason for the use of the term "generally", I do think it's important to keep something like this in

Reply:
FWIW, I like that Lee's wording here provides that extra context.

Reply:
Makes sense to me

Reply:
That makes sense to me 👍